torch/onnx/symbolic.py

```diff
@@ -1488,3 +1488,23 @@ def log_sigmoid(g, input):
 @parse_args('v')
 def erf(g, input):
     return g.op('Erf', input)
+
+
+@parse_args('v', 'i', 'i')
+def flatten(g, input, start_dim, end_dim) :
+    input_dims = input.type().sizes()
+
+    if end_dim < 0 :
```

Comment:
We can emit `Flatten` op in the end_dim=-1 case.

Reply:
it seems to me like not all end_dim=-1 cases can be addressed by the Flatten op.
ex : 
input shape : (1,2,3,4)
output = flatten(input,start_dim=2,end_dim=-1)

expected output shape : (1,2,12)
onnx output shape with flatten: (1,24)  (or other 2D shape, depending on the axis chosen)


But the Flatten op can be used when the output shape is 2D. So given the input shape and start_dim/end_dim, we would have to compute the expected output shape, and use Flatten op if the output shape is 2D.

am I missing something?

Reply:
yeah, you are right, I meant we can emit `Flatten` when possible.